### ISSUE TITLE
Add ability to backprop errors to input (for model interrogation)

### ISSUE THREAD

**Issue by turambar** [deeplearning4j/deeplearning4j#2866]
Placeholder issue. Will add more detailed description later, but TL;DR: we'd like to be able to ask the model what kind of inputs (e.g., image) will maximize the output (e.g., probability that image is cat) and solve for this using gradient descent in the input space (vs. weight space during learning).

**Comment by turambar** [comment]
@AlexDBlack: as we discussed previously

**Comment by raver119** [comment]
Is this issue still relevant?

**Comment by AlexDBlack** [comment]
It's almost there, probably needs a "no exception on null" for this method, plus needs to be tested.
https://github.com/deeplearning4j/deeplearning4j/blob/491507004a571ba0793fac90548a1d8515fd8e63/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/multilayer/MultiLayerNetwork.java#L2589-L2595
